bin/release.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+
+set -e
+
+cd "$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+
+if [[ ! -z "$(git status --porcelain)" ]]; then
+  >&2 echo "ERROR: your working copy is dirty"
+  exit 1
+fi
+
+branch=$(git rev-parse --abbrev-ref HEAD)
+if [[ "$branch" != "master" ]]; then
+  >&2 echo "ERROR: you are not on the master branch"
+  exit 1
+fi
+
+ghr=$(which ghr)
+if [[ -z "$ghr" ]]; then
```

Comment:
which returns non-zero if it fails, so you could just query `$?`

Reply:
Good point. There's lots of issues with `which` (https://stackoverflow.com/questions/592620/how-can-i-check-if-a-program-exists-from-a-bash-script) and I think I used it here out of laziness. I'll update this.